Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -282,34 +316,36 @@ void AVIDump::AddFrame(const u8* data, int width, int height, int stride, const
     s_last_pts = pts_in_ticks;
     error = SendFrameAndReceivePacket(s_codec_context, &pkt, s_scaled_frame, &got_packet);
   }
-  while (!error && got_packet)
+  if (!error && got_packet)
   {
-    // Write the compressed frame in the media file.
-    if (pkt.pts != (s64)AV_NOPTS_VALUE)
-    {
-      pkt.pts = av_rescale_q(pkt.pts, s_codec_context->time_base, s_stream->time_base);
-    }
-    if (pkt.dts != (s64)AV_NOPTS_VALUE)
-    {
-      pkt.dts = av_rescale_q(pkt.dts, s_codec_context->time_base, s_stream->time_base);
-    }
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(56, 60, 100)
-    if (s_codec_context->coded_frame->key_frame)
-      pkt.flags |= AV_PKT_FLAG_KEY;
-#endif
-    pkt.stream_index = s_stream->index;
-    av_interleaved_write_frame(s_format_context, &pkt);
-
-    // Handle delayed frames.
-    PreparePacket(&pkt);
-    error = ReceivePacket(s_codec_context, &pkt, &got_packet);
+    WritePacket(pkt);
   }
   if (error)
     ERROR_LOG(VIDEO, "Error while encoding video: %d", error);
 }
 
 void AVIDump::Stop()
 {
+  AVPacket pkt;
+  int error, got_packet;
+
+  while (true)
+  {
+    // Handle delayed frames.
+    PreparePacket(&pkt);
+    error = ReceivePacket(s_codec_context, &pkt, &got_packet);
```

Comment:
`error` can have its declaration joined with this assignment. `got_packet` can also be moved into this scope as well.